### PR TITLE
Fix/soc 5683

### PR DESCRIPTION
**Pull request by trongtt**

### CODE REVIEW

**Comment by trongtt** [comment]
Yes it should point to the one in platform-ui, I was trying to do that without success. I will look at it more to find a solution.

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 1]
```diff
@@ -242,7 +245,27 @@ public Response getUserAvatarById(@Context UriInfo uriInfo,
     builder.cacheControl(cc);
     return builder.cacheControl(cc).build();
   }
-  
+
+  private Response getDefaultUserAvatar(String avatarUrl) {
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: The case default=404 is not handled. It should return an error 404 if the "default" param equals 404.

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 2]
```diff
@@ -242,7 +245,27 @@ public Response getUserAvatarById(@Context UriInfo uriInfo,
...
+    try {
+      if (avatarUrl != null) {
+        URL url = new URL(avatarUrl);
+        String type = url.openConnection().getHeaderField("Content-Type");
+        if (type != null && type.startsWith("image/")) {
+          InputStream input = url.openStream();
+          return Response.ok(input, type).build();
+        }
+      }
+    } catch (IOException ignored) {
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: Avoid empty catch block. We should at least add a log.debug IMO.

Comment (trongtt) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: fixed

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java (hunk 1 above)
Comment (trongtt) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: fixed

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 3]
```diff
@@ -61,17 +67,21 @@
...
   public static enum ACTIVITY_STREAM_TYPE {
     all, owner, connections, spaces
   }
+
+  private static final Log LOG = ExoLogger.getLogger(UserRestResourcesV1.class);
   
   public UserRestResourcesV1(UserACL userACL, IdentityManager identityManager) {
     this.userACL = userACL;
     this.identityManager = identityManager;
+    ClassLoader test = Thread.currentThread().getContextClassLoader();
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: Useless line ?

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 4]
```diff
@@ -207,42 +217,74 @@ public Response getUserById(@Context UriInfo uriInfo,
           @ApiResponse (code = 400, message = "Invalid query input") })
   public Response getUserAvatarById(@Context UriInfo uriInfo,
                                     @Context Request request,
-                                    @ApiParam(value = "User name", required = true) @PathParam("id") String id) throws IOException {
+                                    @ApiParam(value = "User name", required = true) @PathParam("id") String id,
+                                    @ApiParam(value = "Default avatar", required = false) @QueryParam("default") String defaultAvatar) throws IOException {
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: Please add more detail in the description of this param (in the "value" field) :
* the expected format of the parameter (URL)
* the 404 value which allows to return a 404 http code

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 5]
```diff
@@ -61,17 +67,21 @@
 @Api(tags = VersionResources.VERSION_ONE + "/social/users", value = VersionResources.VERSION_ONE + "/social/users", description = "Operations on users with their activities, connections and spaces")
 public class UserRestResourcesV1 implements UserRestResources {
 
+  private static final String DEFAULT_PROFILE_IMG = "/UserAvtDefault.png";
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: Useless variable.

Comment (trongtt) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: removed

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java (hunk 3 above)
Comment (trongtt) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: yes, removed it

**Pull request merged**